sway/desktop/render.c

```diff
@@ -119,6 +153,7 @@ static void render_surface_iterator(struct sway_output *output, struct sway_view
 	wlr_matrix_project_box(matrix, &box, transform, rotation,
 		wlr_output->transform_matrix);
 
+	set_scale_filter(wlr_output, texture, oc->scale_filter);
```

Comment:
We should probably call `set_scale_filter` from `render_texture`

Reply:
The logic was originally inside `render_texture`, but I moved it out when I switched to a per-output config. My reasoning was:

1. `render_texture` doesn't already maintain a reference to the relevant sway_output
2. `set_scale_filter` is only required, I think, when rendering a texture that corresponds to an application buffer, so I call it before those calls to `render_texture` and do not call it in others such as those in `render_titlebar`.